src/citrine/informatics/reports.py

```diff
@@ -1,18 +1,22 @@
 """Tools for working with reports."""
-from typing import Optional, Type
+from typing import Optional, Type, List, Dict, TypeVar, Iterable, Any
+from abc import abstractmethod
+from itertools import groupby
 
 from citrine._serialization import properties
 from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
 from citrine._serialization.serializable import Serializable
 from citrine._session import Session
+from citrine.informatics.descriptors import Descriptor
+
+SelfType = TypeVar('SelfType', bound='Report')
 
 
 class Report(PolymorphicSerializable['Report']):
```

Comment:
I think the assumption when this class was made was that there would be multiple types of `Report` with some type key to differentiate them. That is not the case now and I don't know if we have any other types of `Report` planned. That said, I could imagine a design space report. But will there ever be a situation in which you don't know what type of report you're getting? I don't think so.

Reply:
Sounds like we should ax it then?

Reply:
yeah

Reply:
Actually, the client's structure for fetching reports makes this less easy. There's a generic `ReportResource` that knows to get from `project/:id/modules/:id/report` but doesn't specify which type of module. It then calls the generic `Report.build` and counts on `Report.build` to figure out what type of `Report` it is.

I could have the module that's calling `ReportResource.get()` pass in an argument that says "this report is going to be a `PredictorReport`" and then it knows to call `PredictorReport.build()` but I think it's more robust to have the serialized report declare what type of report it is, rather than let the client infer. So I think we should stick with the `PolymorphicSerializable` pattern and have a `type` field to `Report` objects.

Reply:
I think that design deserves a re-visit at some point, but certainly not blocking.